### PR TITLE
Add authorized_keys to hosts

### PR DESCRIPTION
**Pull request by mdellweg**

### CODE REVIEW

File: provisioning_templates/snippet/ssh_authorized_keys.erb [hunk 1]
```diff
@@ -0,0 +1,38 @@
...
+
+ssh_authorized_keys: public keys to be put in /root/.ssh/authorized_keys
+
+This template sets up SSH keys in any host so that as long as your public
+SSH key is in ssh_authorized_keys, you can SSH into a host.
+%>
+
+<% if !host_param('ssh_authorized_keys').blank? %>
+<% authorized_keys = host_param('ssh_authorized_keys') %>
+<% ssh_user = 'root' %>
```
Comment (ezr-ondrej) [inline provisioning_templates/snippet/ssh_authorized_keys.erb]: does ssh user has to be root?

**Review by ezr-ondrej — COMMENTED** [review]
I believe this would be an awesome addition, but I feel like it would make sense while at it to make ti possible to at least choose the user. The best would be if we could somehow have keys for multiple users. The biggest issue is the potential risk that we add the keys to `root` user, but users may have already have some `ssh_authorized_keys` param for some other automation (Ansible i.e.).
Apart of it the snippet does basically the same thing as `remote_execution_ssh_keys` it would be great if we could somehow unify it into one.

Does it make sense? Or is there even a value to have it for `root`? Any thoughts?

File: provisioning_templates/snippet/ssh_authorized_keys.erb (hunk 1 above)
Comment (mdellweg) [reply provisioning_templates/snippet/ssh_authorized_keys.erb]: The user could be configurable. Sure. The remote_execution_ssh_keys adds sudo to the corresponding user if needed. Also i think there is a way to add the ssh_key of the foreman user into the newly created machine. But, i think, all of those serve different purposes.

I tried to mimic the logic of the `ssh_authorized_keys` parameter as used here:
https://github.com/theforeman/community-templates/blob/develop/provisioning_templates/provision/coreos_provision.erb#L32
And bring it to more platforms.

File: provisioning_templates/snippet/ssh_authorized_keys.erb [hunk 2]
```diff
@@ -0,0 +1,38 @@
...
+<% ssh_path = "~#{ssh_user}/.ssh" %>
```
Comment (ekohl) [inline provisioning_templates/snippet/ssh_authorized_keys.erb]: I think `~#{ssh_user}` is a bashism and probably doesn't work on other shells.

File: provisioning_templates/snippet/ssh_authorized_keys.erb [hunk 3]
```diff
@@ -0,0 +1,38 @@
...
+
+Parameters:
+
+ssh_authorized_keys: public keys to be put in /root/.ssh/authorized_keys
+
+This template sets up SSH keys in any host so that as long as your public
+SSH key is in ssh_authorized_keys, you can SSH into a host.
+%>
+
+<% if !host_param('ssh_authorized_keys').blank? %>
```
Comment (ekohl) [inline provisioning_templates/snippet/ssh_authorized_keys.erb]:
```suggestion
<% unless host_param('ssh_authorized_keys').blank? -%>
```

File: provisioning_templates/snippet/ssh_authorized_keys.erb [hunk 4]
```diff
@@ -0,0 +1,38 @@
+<%#
+kind: snippet
+name: ssh_authorized_keys
+model: ProvisioningTemplate
+snippet: true
+%>
+
```
Comment (ekohl) [inline provisioning_templates/snippet/ssh_authorized_keys.erb:7]: This generates a lot of whitespace. Could you have a look at the various blocks?

**Review by ekohl — COMMENTED** [review]
We also have https://github.com/theforeman/community-templates/blob/develop/provisioning_templates/snippet/create_users.erb which is a more native way of handling SSH keys. Any reason to prefer a host param?

**Comment by mdellweg** [comment]
@ekohl With host_params it can be set on the hostgroup level or even specific to domains or just about any other organization structure of foreman.
Also as i tried to express above the parameter `ssh_authorized_keys` is already used on the some cloud provisioning templates, and i think it would be nice to handle it in a consistent way across all platforms.

File: provisioning_templates/snippet/ssh_authorized_keys.erb (hunk 2 above)
Comment (mdellweg) [reply provisioning_templates/snippet/ssh_authorized_keys.erb]: Not invented here... would you prefer calling `getent passwd #{ssh_user} | cut -d":" -f 6`?
https://github.com/theforeman/community-templates/blob/03281e9138e178f336958d1ef83739483c048078/provisioning_templates/snippet/create_users.erb#L12

Comment (ekohl) [reply provisioning_templates/snippet/ssh_authorized_keys.erb]: I guess it's safe enough and otherwise patches welcome :)

**Comment by tbrisker** [comment]
On behalf of the Foreman team, thank you for your contribution to the Foreman community templates repository! 
This repository is being archived and all templates are migrated to the relevant repositories. If you wish to pursue this change, please re-open the PR against the relevant repository. For more information and background, please [click here]( https://community.theforeman.org/t/heads-up-community-templates-repo-will-be-archived-tomorrow/19539).

**Pull request closed without merging**